### ISSUE TITLE
Error notifcation does not show on CRUD server error

### ISSUE THREAD

**Issue by mikkelfish** [marmelab/admin-on-rest#365]
Hello, I noticed that the notification does not pop up when there is a server error. After looking at the code I have identified the problem:

In crudFetch, it properly catches the error from the restClient and then has
```
{
    type: `${type}_FAILURE`,
    error: error.message ? error.message : error,
    requestPayload: payload,
    meta: { ...meta, fetchResponse: restType, fetchStatus: FETCH_ERROR },
 }
```

Notice that the error will be a string, since it just pulls out error.message.

However, in crudResponse it has 
`put(showNotification(error.message, 'warning'))`

Which is null because error is now a string. It just needs to be in sync: either don't pull out the message in crudFetch or don't reference message in crudResponse, and then it works.

Thanks!

**Comment by matteolc** [comment]
Hi
I agree, errors should/can be a bit more structured than a plain string.
For instance, with a rails api, create/update errors are sent back from the api as a hash, where each key represents a field that gave an error, and the value is the error.
This can be very useful in displaying create/update errors directly in the form fields, after posting the form, when validations are handled by the server. Some validations can be handled by the client (for instance required values), but some other complex validations are typically handled by the server. It's not pretty to duplicate validations both in the client and the server, even though I agree it can be more efficient (but again only with basic validations: required, number, email; with complex validations, the duplication of the logic in the client and server defeats the efficiency gained in having less network requests). In the following image, the 'can't be blank' message is shown after posting the form and getting a response different than 201 (created) or 200 (ok, updated).
It would be nice to pass on the error messages returned by the server also the components, or even better store them in a ... store.
Thanks

![error](https://cloud.githubusercontent.com/assets/78274/23329862/20a9d0d6-fb41-11e6-8b87-b9fefa5e5ef1.png)

**Comment by matteolc** [comment]
Of course in this case one would deal with i18n also at server side. Rails has a i18n mechanism which is very similar to the one adopted by aor, so it all fits nicely. I have the (aor http) client send the locale set in aor, and the server replies with the locale requested by the client. So my server error messages are also translated in line with the client locale.
Thanks

**Comment by fzaninotto** [comment]
Already fixed in #361.

As for server-side validation, it was already asked in #97
